mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -3330,6 +3330,27 @@ func TestAccContainerCluster_withDNSConfig(t *testing.T) {
 	})
 }
 
+func TestAccContainerCluster_withGatewayApiConfig(t *testing.T) {
+	t.Parallel()
+
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withGatewayApiConfig(clusterName, "CHANNEL_STANDARD"),
```

Comment:
If the field is updatable: Please add a second set of steps here that uses a different channel, and make sure to handle the new fields in the Update() function.

Otherwise: please mark `channel` as `ForceNew`

Reply:
I'll write two more tests!
One for adding the gateway api to a cluster that hasn't used it before.
and another one that tests to switch between the channels.

That should suffice :)

Reply:
I added a test case with multiple steps:
 - Creating a cluster with the standard channel for the gateway api
 - Updating the cluster to use the experimantel channel

Another test case has been added for checking against invalid channel names.